client/my-sites/domains/domain-management/dns/caa-record.jsx

```diff
@@ -0,0 +1,103 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import React from 'react';
+import classnames from 'classnames';
+import { localize } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import FormFieldset from 'components/forms/form-fieldset';
+import FormInputValidation from 'components/forms/form-input-validation';
+import FormLabel from 'components/forms/form-label';
+import FormTextarea from 'components/forms/form-textarea';
+import FormTextInputWithAffixes from 'components/forms/form-text-input-with-affixes';
+import FormSelect from 'components/forms/form-select';
+import FormTextInput from 'components/forms/form-text-input';
+
+class CaaRecord extends React.Component {
+	static propTypes = {
+		fieldValues: PropTypes.object.isRequired,
+		onChange: PropTypes.func.isRequired,
+		selectedDomainName: PropTypes.string.isRequired,
+		show: PropTypes.bool.isRequired,
+	};
+
+	getValidationErrorMessage( value ) {
+		const { translate } = this.props;
+
+		if ( value?.length === 0 ) {
+			return translate( 'CAA record may not be empty' );
+		}
+
+		return null;
+	}
+
+	render() {
+		const { fieldValues, isValid, onChange, selectedDomainName, show, translate } = this.props;
+		const classes = classnames( { 'is-hidden': ! show } );
+		const isNameValid = isValid( 'name' );
+		const isDataValid = isValid( 'data' );
+		const dataValidationError = this.getValidationErrorMessage( fieldValues.data );
+
+		const options = [ 'issue', 'issuewild' ].map( ( type ) => {
+			return (
+				<option key={ type } value={ type }>
+					{ type }
+				</option>
+			);
+		} );
+
+		return (
+			<div className={ classes }>
+				<FormFieldset>
+					<FormLabel>{ translate( 'Name', { context: 'Dns Record' } ) }</FormLabel>
+					<FormTextInputWithAffixes
+						name="name"
+						placeholder={ translate( 'Enter subdomain (optional)', {
+							context:
+								'Placeholder shown when entering the optional subdomain part of a new DNS record',
+						} ) }
+						isError={ ! isNameValid }
+						onChange={ onChange }
+						value={ fieldValues.name }
+						suffix={ '.' + selectedDomainName }
+					/>
+					{ ! isNameValid && <FormInputValidation text={ translate( 'Invalid Name' ) } isError /> }
+				</FormFieldset>
+
+				<FormFieldset>
+					<FormLabel>{ translate( 'Tag', { context: 'Dns Record' } ) }</FormLabel>
+
+					<FormSelect name="tag" onChange={ onChange } value={ fieldValues.tag }>
+						{ options }
+					</FormSelect>
+				</FormFieldset>
+
+				<FormFieldset>
+					<FormLabel>{ translate( 'Flags', { context: 'Dns Record' } ) }</FormLabel>
```

Comment:
:information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.